### ISSUE TITLE
implement `FineTuningValidator`

### ISSUE THREAD

**Issue by wochinge** [RasaHQ/rasa#9335]
**Overview of the Solution**:
We need to implement the `FineTuningValidator`. This is a component at the beginning of the graph which receives all training data and raises an exception in case `is_finetuning=True` and finetuning is not possible (e.g. because new labels were added). In case we are doing a regular training (and not finetuning) this persists the necessary information extracted from the training data to be able to validate when initialized via `load` whether we can finetune.

The check whether the model is finetunable should be loosely based upon [this current code piece](https://github.com/RasaHQ/rasa/blob/f78a6a17d5146248500630a2397ee099a24e0f5c/rasa/model.py#L548).

[This figure](https://www.notion.so/rasa/Rasa-Open-Source-3-0-Architecture-Implementation-Proposal-51ab90b05c41435ca98189a101676a1e#24ef56343a554fe88748651864fe9e77) shows the component's position in the graph.

**Input to the component**
- `TrainingDataImporter`

**Output:** 
* `None`
* raises an exception if finetuning can't be done

**Blockers**
* https://github.com/RasaHQ/rasa/pull/9184

**Definition of Done**:
- [ ] Component is implemented and unit tested
